error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/OptionalRules.java

```diff
@@ -100,6 +100,20 @@ Function<Optional<T>, T> after() {
     }
   }
 
+  /** Prefer {@link Optional#equals(Object)} over more contrived alternatives. */
+  static final class OptionalHasValue<T, S> {
```

Comment:
Hmm, I'm not sure if I agree with the name based on our naming algorithm 🤔.

`OptionalEquals` or `OptionalEqualsOptional` seems more accurate, WDYT?

Reply:
I guess even `OptionalEqualsOptionalOf`; but this work for me for now :+1: